### PR TITLE
Parse and store specials with unique ptr

### LINKED ISSUE #1546: SpecialsParser should return unique_ptr and not leak memory.
**Environment: all**

**Description**
The `parse/SpecialsParser.cpp` is a generator.  In C++11 a generator should return a `std::unique_ptr<>` to indicate that it has created something that requires memory management and that it is passing management responsibility to the calling function.  The calling function can chose to receive the pointer as a `unique_ptr<>` or transfer it to a `shared_ptr<>` if it needs a shared ownership model.

Furthermore the error condition near line 29 leaks the passed in pointer.  This is a static leak, it only leaks objects that have failed to parse once per game.

**Expected Result**
Return `std::unique_ptr<>`s.

**Steps to reproduce**
Play game.

**Suggestions**
A solution of the same issue in the BuildingType parser is in #1544, is anyone would like to assign themselves and help out.

### PR DESCRIPTION
**Pull request by dbenage-cx**
On local system, compiling with boost v1.62 or lower fails(1.63+ is ok), opening PR for CI results to help troubleshoot

Resolves #1546

### CODE REVIEW

**Comment by LGM-Doyle** [comment]
@dbenage-cx I can confirm that this works on Fedora with Boost v1.63.

**Comment by dbenage-cx** [comment]
1.63 introduces variadic templates, the same results can be seen by defining `BOOST_NO_CXX11_VARIADIC_TEMPLATES`.

I was not able to determine the correct macros and order to increase number of args.
Instead, using a struct to handle transition.

File: parse/SpecialsParser.cpp [hunk 1]
```diff
@@ -8,29 +8,67 @@
...
+    inline ostream& operator<<(ostream& os, const std::pair<const std::string, std::unique_ptr<Special>&) { return os; }
 }
 #endif
 
 namespace {
     const boost::phoenix::function<parse::detail::is_unique> is_unique_;
 
-    const boost::phoenix::function<parse::detail::insert> insert_;
+    struct special_pod {
+        special_pod (std::string name_,
```
Comment (geoffthemedio) [inline parse/SpecialsParser.cpp]: remove space after `pod`

**Comment by geoffthemedio** [comment]
otherwise looks fine

**Pull request merged**